src/ServiceControl.Monitoring/QueueLength/QueueLengthCalculator.cs

```diff
@@ -3,45 +3,80 @@
     using System;
     using System.Collections.Concurrent;
     using System.Collections.Generic;
+    using System.Linq;
 
     class QueueLengthCalculator : IQueueLengthCalculator
     {
-        public void UpdateReceivedSequence(string key, long value, string queue)
+        public bool UpdateReceivedSequence(string endpointName, string key, double value, string queue)
         {
-            receivedSequences.AddOrUpdate(key, _ => value, (_, prev) => Math.Max(prev, value));
-            sessionKeyToQueueMapping.GetOrAdd(key, _ => queue);
+            var virtualQueue = new VirtualQueueId
```

Comment:
Would it make sense to just pass in a VirtualQueueId here? The callers of abstraction know about it because other methods on this class return it.